maintenance/Maintenance.php

```diff
@@ -1100,10 +1100,10 @@ protected static function getCoreScripts() {
 	 */
 	protected function &getDB( $db, $groups = array(), $wiki = false ) {
 		if ( is_null( $this->mDb ) ) {
-			return wfGetDB( $db, $groups, $wiki );
-		} else {
-			return $this->mDb;
+			$this->mDb = wfGetDB( $db, $groups, $wiki );
 		}
+
+		return $this->mDb;
```

Comment:
Are we sure that no script makes a check whether `$this->mDb`  is not set and performs a different action assuming that `wfGetDB` was used?

That would be pretty hacky, but let's double check it.